src/index.js

```diff
@@ -36,65 +42,74 @@ class WebSdkApi extends EventEmitter {
     if (authFormCustomization !== undefined) {
       if (authFormCustomization.logoUrl !== undefined) {
-        this.chaperone_url.searchParams.set("logo_url", makeUrlAbsolute(authFormCustomization.logoUrl))
+        final_chaperone_url.searchParams.set("logo_url", makeUrlAbsolute(authFormCustomization.logoUrl))
       }
       if (authFormCustomization.appName !== undefined) {
-        this.chaperone_url.searchParams.set("app_name", authFormCustomization.appName)
+        final_chaperone_url.searchParams.set("app_name", authFormCustomization.appName)
       }
       if (authFormCustomization.infoLink !== undefined) {
-        this.chaperone_url.searchParams.set("info_link", authFormCustomization.infoLink)
+        final_chaperone_url.searchParams.set("info_link", authFormCustomization.infoLink)
       }
       if (authFormCustomization.publisherName !== undefined) {
-        this.chaperone_url.searchParams.set("publisher_name", authFormCustomization.publisherName)
+        final_chaperone_url.searchParams.set("publisher_name", authFormCustomization.publisherName)
       }
       if (authFormCustomization.anonymousAllowed !== undefined) {
-        this.chaperone_url.searchParams.set("anonymous_allowed", authFormCustomization.anonymousAllowed)
+        final_chaperone_url.searchParams.set("anonymous_allowed", authFormCustomization.anonymousAllowed)
       }
       if (authFormCustomization.registrationServer !== undefined) {
-        this.chaperone_url.searchParams.set("registration_server_url", makeUrlAbsolute(authFormCustomization.registrationServer.url))
-        this.chaperone_url.searchParams.set("registration_server_payload", JSON.stringify(authFormCustomization.registrationServer.payload))
+        final_chaperone_url.searchParams.set("registration_server_url", makeUrlAbsolute(authFormCustomization.registrationServer.url))
+        final_chaperone_url.searchParams.set("registration_server_payload", JSON.stringify(authFormCustomization.registrationServer.payload))
       }
       if (authFormCustomization.skipRegistration !== undefined) {
-        this.chaperone_url.searchParams.set("skip_registration", authFormCustomization.skipRegistration)
+        final_chaperone_url.searchParams.set("skip_registration", authFormCustomization.skipRegistration)
       }
     }
 
     let child
     try {
-      child = await COMB.connect(this.chaperone_url.href, 60000);
+      child = await COMB.connect(final_chaperone_url.href, 60000);
     } catch (err) {
       if (err.name === "TimeoutError")
         console.log("Chaperone did not load properly. Is it running?");
       throw err;
     }
 
+    const webSdkApi = new WebSdkApi(child);
+
+    // based on discussed model, chaperone would either return null or an error message (string)
+    const error_message = await child.call("handshake")
+    if (error_message) {
+      webSdkApi.iframe.display = "none"
+      throw new Error(error_message)
+    }
```

Comment:
this has to go below line 91 `webSdkApi.iframe = document.getElementsByClassName("comb-frame-0")[0];`

Or you'll get an error trying to access `.iframe` on websdk

I think this error check can go last thing in the function right before returning the web-sdk object

Reply:
good catch.